### PR TITLE
Gui/logger

### PR DESCRIPTION
**Pull request by RivkiYagelnik**
GUI : Registering for the logger and using it in MainWindow

### CODE REVIEW

File: gui/src/main_window.cpp [hunk 1]
```diff
@@ -14,12 +14,15 @@
...
 
 int sizeSquare = 120;
+logger MainWindow::guiLogger("gui");
 
 MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), timer(nullptr)
 {
     QWidget *centralWidget = new QWidget(this);
     setCentralWidget(centralWidget);
 
+    MainWindow::guiLogger.logMessage(logger::LogLevel::ERROR, "MainWindow::MainWindow()  Log file successfully read");
```
Comment (ChedvaErblich) [inline gui/src/main_window.cpp]: is it an error? I see your message says "Log file successfully read" but you use the ERROR level

File: gui/src/main_window.cpp [hunk 2]
```diff
@@ -560,8 +606,14 @@ void MainWindow::createProcessConfigFile(int id, const QString &processPath) {
         configFile.write(jsonDoc.toJson());
         configFile.close();
         qDebug() << "Config file created at:" << filePath;
+
+        // Log the successful creation of the config file
+        guiLogger.logMessage(logger::LogLevel::INFO, "Config file created at: " + filePath.toStdString());
     } else {
         qWarning() << "Failed to create config file at:" << filePath;
+
+        // Log the failure to create the config file
```
Comment (ChedvaErblich) [inline gui/src/main_window.cpp]: I think you can remove the comment, it's clear enough.

File: gui/src/main_window.cpp [hunk 3]
```diff
@@ -560,8 +606,14 @@ void MainWindow::createProcessConfigFile(int id, const QString &processPath) {
         configFile.write(jsonDoc.toJson());
         configFile.close();
         qDebug() << "Config file created at:" << filePath;
+
+        // Log the successful creation of the config file
```
Comment (ChedvaErblich) [inline gui/src/main_window.cpp]: I think you can remove the comment, it's clear enough.

File: gui/include/draggable_square.h [hunk 4]
```diff
@@ -7,13 +7,11 @@
...
-
-class DraggableSquare : public QWidget
-{
+class DraggableSquare : public QWidget {
     Q_OBJECT
 
-   public:
-   friend class DraggableSquareTest;
+public:
+    friend class DraggableSquareTest;
```
Comment (ChedvaErblich) [inline gui/include/draggable_square.h:14]: White space :)

File: gui/src/frames.cpp [hunk 5]
```diff
@@ -90,6 +101,11 @@ void Frames::updateFrames()
         }
     }
 
+    // Log the number of remaining active log entries
```
Comment (ChedvaErblich) [inline gui/src/frames.cpp]: Same here, you can remove the comment

File: gui/test/test_main_window.cpp [hunk 6]
```diff
@@ -17,21 +16,21 @@ private slots:
     void addId(int id);
 };
 
-
 void TestMainWindow::testCreateNewProcess()
 {
     MainWindow window;
-    int newProcessId = 6; // Ensure this is greater than 5 and unique
+    int newProcessId = 6;  // Ensure this is greater than 5 and unique
```
Comment (ChedvaErblich) [inline gui/test/test_main_window.cpp]: white space

**Comment by ChedvaErblich** [comment]
Great job!

One thing,
I noticed many coding convention changes along with adding the log messages.
These should be split into two separate commits.

File: gui/src/main_window.cpp (hunk 1 above)
Comment (RivkiYagelnik) [reply gui/src/main_window.cpp]: This is unnecessary code, I wrote it for the logger experiment.
Tin what to record there for the logger.
That's why I left it out...

File: gui/src/main_window.cpp (hunk 2 above)
Comment (RivkiYagelnik) [reply gui/src/main_window.cpp]: I left it out and i put insted a sentance to the logger.
why do you think it's not good?

File: gui/src/main_window.cpp (hunk 3 above)
Comment (RivkiYagelnik) [reply gui/src/main_window.cpp]: that too..

File: gui/src/main_window.cpp (hunk 1 above)
Comment (ChedvaErblich) [reply gui/src/main_window.cpp]: Still, if it's not an error please use log debug

**Pull request closed without merging**